src/markermap/utils.py

```diff
@@ -522,7 +560,17 @@ def __init__(self, input_size, hidden_layer_size, z_size, output_size = None, bi
         self.encoder, self.enc_mean, self.enc_logvar = make_encoder(input_size,
                 hidden_layer_size, z_size, bias = bias, batch_norm = batch_norm)
 
-        self.decoder, self.dec_logvar = make_gaussian_decoder(output_size, hidden_layer_size, z_size, bias = bias, batch_norm = batch_norm)
+        if decoder is None:
+            decoder = make_gaussian_decoder(
+                output_size,
+                hidden_layer_size,
+                z_size,
+                bias = bias,
+                batch_norm = batch_norm,
+            )
+
+        self.decoder = decoder
+        self.dec_logvar = deepcopy(decoder)
```

Comment:
I think this might introduce a bug, or actually a new feature.

This ties the initial weights of the two decoders, whereas that is not the case in the original. Not sure if an improvement or not, but it might cause issues with reproducing results.

Reply:
Ah, that is a good catch. We probably want to avoid that. I guess I will just allow users to set decoder and dec_logvar separately.